.bazelrc

```diff
@@ -1,2 +1,13 @@
-try-import %workspace%/user.bazelrc
-build --cxxopt='-std=c++17'
+# Workaround for Bazel sync and proper intellisense on Linux
+build --action_env=BAZEL_CXXOPTS="-std=c++17"
+
+# Default build compiler options
+# TODO: Consider to add Windows and Ubuntu specific build flavors
+build --cxxopt="-std=c++17"
+build --cxxopt="-Werror"
+build --cxxopt="-Wall"
+build --cxxopt="-Wextra"
+build --cxxopt="-Wpedantic"
+# This is needed to suppress warnings introduced by stb_image.h
+build --cxxopt="-Wno-sign-compare"
+build --cxxopt="-Wno-unused-but-set-variable"
```

Comment:
I don't like these being global, different packages may require different flags. This also breaks Windows build since Windows requires glad (an extern lib) and glad can't be built with -Werror since it contains warnings (at least I suspect that's the reason, I did not investigate to the end). Also I definitely don't want to use -Wno-sign-compare and -Wno-unused-but-set-variable for all packages, these were added to extern package because stb_image lib couldn't be built without them.

Reply:
Ah, yes. I was unable to test the Windows build as I don't have MSYS installed. Then again, it's better to have the project build with the `msvc` compiler as this is more of a Windows native compiler. Also, we could introduce `linux` and `windows` build configurations that allow us to have specific rules for each OS.

Also, I agree with the `-Wno-sign-compare` and `-Wno-unused-but-set-variable`, but in the `extern` folder I would create separate `WORKSPACE` Bazel packages so we have isolated libraries and are not clumped together in a single rule/target.